var/spack/repos/builtin/packages/py-nptyping/package.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyNptyping(PythonPackage):
+    """Type hints for numpy"""
+    homepage = "https://github.com/ramonhagenaars/nptyping"
+    pypi = "nptyping/nptyping-2.4.1.tar.gz"
+
+    version("2.4.1", sha256="57ba684ee5fc5eb681ee04270ee94adb879e4372ce6b640defa08ace8e1df295")
+    version(
```

Comment:
Is there any reason to add these ancient versions that don't have sdists on PyPI?

Reply:
I'm mostly upstreaming changes that have been lying around here for a long time, some of these ancient versions are, apparently, still being used here. I've added the latest version specifically for this PR (although I admit I should've tested properly).

On a related note: as I have more that I'd like to upstream, is it better to make a bunch of smaller PRs with packages that relate to one another, or reduce the PR-spam a little by making a bigger one that contains unrelated packages?

Reply:
We try to follow a one-package-per-PR system, although it's fine to combine 2+ packages if the changes to one package depend on another.